### PR TITLE
Solved issues with AWS CentOS 7.

### PR DESCRIPTION
**Pull request by jdelaporte**
Hi Greg,

Thanks for publishing this role on GitHub! 

I had some issues running it on an AWS instance with CentOS 7 installed. On my machine, hostname module didn't work. Also, firewalld was not installed by default. I made a couple changes that worked for me, so I am sending them back to you in case you want to incorporate them

Thanks,
Joanna

### CODE REVIEW

File: tasks/firewalld.yml~ [hunk 1]
```diff
@@ -0,0 +1,28 @@
+---
```
Comment (gregswift) [inline tasks/firewalld.yml~]: Can you remove this file from your commit?

File: tasks/main.yml~ [hunk 2]
```diff
@@ -0,0 +1,49 @@
+---
+#- name: Hard set the hostname
```
Comment (gregswift) [inline tasks/main.yml~]: can you remove this file from your PR

File: tasks/main.yml [hunk 3]
```diff
@@ -2,6 +2,8 @@
 - name: Hard set the hostname
   hostname: name={{ ansible_fqdn }}
 
+# - shell: hostname {{ ansible_fqdn }}
```
Comment (gregswift) [inline tasks/main.yml]: Can you remove this?

Comment (jdelaporte) [inline tasks/main.yml]: Sure. Give me a few minutes while I figure out how. I am painfully new to
git pull requests still.

On Mon, Nov 23, 2015 at 4:33 PM, Greg Swift <notifications@github.com>
wrote:

> In tasks/main.yml
> <https://github.com/gregswift/ansible-ipaserver/pull/1#discussion_r45671176>
> :
>
> > @@ -2,6 +2,8 @@
> >  - name: Hard set the hostname
> >    hostname: name={{ ansible_fqdn }}
> >
> > +# - shell: hostname {{ ansible_fqdn }}
>
> Can you remove this?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/gregswift/ansible-ipaserver/pull/1/files#r45671176>.
>



-- 


Joanna Delaporte
Linux Systems Administrator | Parkland College
joannadelaporte@gmail.com

**Comment by jdelaporte** [comment]
I think I cleaned it up successfully. Let me know if it doesn't look right.

**Comment by gregswift** [comment]
Joanna,

I'm down with brining your firewall changes in.  I think i'm gonna turn around and add some *when* logic to them, but they are a great addition.  If you could clean the other bits out that would be awesome.

thanks

**Comment by jdelaporte** [comment]
Hi Greg,

What other bits? Do you mean the extra backup files? Or some comments or
something in the firewall.yml file? I'm offline now, so I can't see it at
the moment.

Thanks!
Joanna

File: tasks/firewalld.yml [hunk 4]
```diff
@@ -1,10 +1,20 @@
...
+- name: Enable and start firewalld
+  service: name=firewalld state=started enabled=yes
+  
+- name: Enable the LDAPS firewalld ruleset provided by freeipa 
   firewalld: service=freeipa-ldaps permanent=true state=enabled
   notify: reload firewalld
   when: ansible_distribution == "Fedora"
 
-- name: Enable the individual rulesets for CentOS
+- name: Enable the individual rulesets for CentOS
```
Comment (gregswift) [inline tasks/firewalld.yml]: some trailing whitespace made it in

**Comment by gregswift** [comment]
The 'other bits' was just the other commetns i made. it all looks great

**Comment by jdelaporte** [comment]
Hi Greg, 

I added a basic RedHat family conditional to the yum install firewalld. You might want to make it more specific. I cleaned out a couple whitespaces I found, too.

File: tasks/firewalld.yml [hunk 5]
```diff
@@ -1,4 +1,14 @@
 ---
+#firewalld doesn't exist on AWS CentOS 7 by default
+- name: Make sure firewalld is installed
+  yum: name=firewalld state=present
+  when: ansible_os_family == "RedHat"
```
Comment (gregswift) [inline tasks/firewalld.yml]: I dont think we need to add the when here. other families have firewalld too, but it was a good thought

Comment (jdelaporte) [inline tasks/firewalld.yml]: Right, but they don't all have yum.

Comment (gregswift) [inline tasks/firewalld.yml]: Touche. Good point.

**Comment by gregswift** [comment]
I've updated the role with a wrapper boolean on the firewalld subtasks.  However I need to test this before i cut a release. not sure when i'll get to that.

**Pull request merged**